pkg/resources/management.cattle.io/v3/globalrolebinding/validator.go

```diff
@@ -172,8 +182,16 @@ func (a *admitter) validateGlobalRole(globalRole *v3.GlobalRole, fieldPath *fiel
 	return nil
 }
 
+// canSkipValidation checks if validation matches one of two scenarios which allow us to skip validation.
+// 1. Object is scheduled for deletion (deletion timestamp set)
```

Comment:
Why do we allow updates if the object is pending deletion?

Reply:
So controllers can remove their finalizers since K8s will block deletes until the finalizers are removed. I will also note that this is safe since you can not remove the deletion timestamp once it has been set.

Reply:
>  I will also note that this is safe since you can not remove the deletion timestamp once it has been set.

Yeah, I was wondering if there was a possibility that somehow (finalizers perhaps) a GRB stays alive (although still in pending delete state) and an update could still cause some security issues. eg: GRB binds to `regular-user`, then is set to be deleted but stays alive for a while, and is updated to bind to `superuser`. Does that make any sense?

Reply:
Yeah, that is a valid concern our controller, which assign permissions based on GRB, check if the deletion timestamp is set before operating on the object.